scripts/demangle.py

```diff
@@ -41,9 +41,7 @@ def extract_func_code(func_name: str, code_content: str, if_c_code: bool):
         warning(
             f"ERROR: Malformed function definition for '{ori_func_name}' in code file"
         )
-        return (
-            f"ERROR: Malformed function definition for '{ori_func_name}' in code file"
-        )
+        return ""
```

Comment:
`None` is a keyword, not the empty string. A empty string would still have type `str`.

Please refer to https://docs.python.org/3/library/typing.html#typing.Optional
https://en.wikipedia.org/wiki/Option_type

Reply:
Since I write code content back to JSON file, I think it needs to return an empty string. Otherwise, that part in json is None.

Reply:
If you get a malformed function, you still need to return `None`. Or otherwise the caller to this function may think it is an empty function like
```c
void foo() {}
```